### PR TITLE
Configuration option for HTTP request header redaction.

### PR DESCRIPTION
**Pull request by ucosty**
Add redact_headers configuration option, which allows specific HTTP request headers to be redacted.

I've run into situations where people have added things like API keys into HTTP headers, which are making their way into our logs.

### CODE REVIEW

**Comment by adriansr** [comment]
Thanks for your contribution!

It looks fine to me, pending CI. 

Can you also add an entry in `CHANGELOG.next.asciidoc` as well as document the new configuration option in `packetbeat-options.asciidoc` ?

**Comment by ucosty** [comment]
@adriansr I've updated the next changelog, as well as the asciidoc as per your request.

File: packetbeat/tests/system/packetbeat.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+from beat.beat import TestCase
 import os
 import sys
 import subprocess
 import json
 
 sys.path.append(os.path.join(os.path.dirname(__file__), '../../../libbeat/tests/system'))
 
-from beat.beat import TestCase
-from beat.beat import Proc
```
Comment (adriansr) [inline packetbeat/tests/system/packetbeat.py]: Can you revert this change? It is causing the tests to fail because its looking for `beat.beat` on the wrong path.

Comment (ucosty) [reply packetbeat/tests/system/packetbeat.py]: All done, and it looks like that build stage is passing now.

**Comment by adriansr** [comment]
jenkins, test this

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -82,6 +82,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Add dns.question.subdomain and dns.question.top_level_domain fields. {pull}14578[14578]
 - Add support for mongodb opcode 2013 (OP_MSG). {issue}6191[6191] {pull}8594[8594]
 - NFSv4: Always use opname `ILLEGAL` when failed to match request to a valid nfs operation. {pull}11503[11503]
+- Added redact_headers configuration option, to allow HTTP request headers to be redacted whilst keeping the header field included in the beat.
```
Comment (adriansr) [inline CHANGELOG.next.asciidoc]:
```suggestion
- Added redact_headers configuration option, to allow HTTP request headers to be redacted whilst keeping the header field included in the beat. {pull}15353[15353]
```

**Comment by adriansr** [comment]
jenkins, test this

**Comment by ucosty** [comment]
@adriansr do you know when this feature might make a release? It doesn't seem to be included in any of the releases since the merge to master.

**Comment by adriansr** [comment]
@ucosty no, it was too late for 7.6.0, but I will backport it so its released in 7.7.0
